athena-db2-as400/src/main/java/com/amazonaws/athena/connectors/db2as400/Db2As400MetadataHandler.java

```diff
@@ -0,0 +1,481 @@
+                while (rs.next()) {
+                    list.add(rs.getString("NAME"));
+                }
+            }
+        }
+        return list;
+    }
+
+    /**
+     * Appropriate data type to arrow type conversions will be done by fetching data types of columns.
+     * This function creates the arrow Schema pojo from jdbc database metadata of the table.
+     *
+     * @param jdbcConnection
+     * @param tableName
+     * @param partitionSchema
+     * @return
+     * @throws Exception
+     */
+    private Schema getSchema(Connection jdbcConnection, TableName tableName, Schema partitionSchema)
+            throws Exception
+    {
+        SchemaBuilder schemaBuilder = SchemaBuilder.newBuilder();
+        try (ResultSet resultSet = getColumns(jdbcConnection.getCatalog(), tableName, jdbcConnection.getMetaData());
+             Connection connection = getJdbcConnectionFactory().getConnection(getCredentialProvider());
+             PreparedStatement stmt = connection.prepareStatement(Db2As400Constants.COLUMN_INFO_QUERY)) {
+            stmt.setString(1, tableName.getSchemaName());
+            stmt.setString(2, tableName.getTableName());
+            try (ResultSet dataTypeResultSet = stmt.executeQuery()) {
+                HashMap<String, String> columnNameMap = new HashMap<>();
+                // fetch data types of columns and prepare map with column name and typeName.
+                while (dataTypeResultSet.next()) {
+                    columnNameMap.put(dataTypeResultSet.getString("COLUMN_NAME"), dataTypeResultSet.getString("DATA_TYPE"));
+                }
+
+                while (resultSet.next()) {
+                    ArrowType columnType = JdbcArrowTypeConverter.toArrowType(
+                            resultSet.getInt("DATA_TYPE"),
+                            resultSet.getInt("COLUMN_SIZE"),
+                            resultSet.getInt("DECIMAL_DIGITS"),
+                            configOptions
+                    );
+                    String columnName = resultSet.getString("COLUMN_NAME");
+                    String typeName = columnNameMap.get(columnName);
+                    /*
+                     * Converting REAL, DOUBLE, DECFLOAT data types into FLOAT8 since framework is unable to map it by default
+                     */
+                    if ("real".equalsIgnoreCase(typeName) || "double".equalsIgnoreCase(typeName) || "decfloat".equalsIgnoreCase(typeName)) {
+                        columnType = Types.MinorType.FLOAT8.getType();
+                    }
+
+                    /*
+                    If arrow type is struct then convert to VARCHAR, because struct is
+                    considered as Unhandled type by JdbcRecordHandler's makeExtractor method.
+                     */
+                    if (columnType != null && columnType.getTypeID().name().equalsIgnoreCase("Struct")) {
+                        columnType = Types.MinorType.VARCHAR.getType();
+                    }
```

Comment:
Is there a reason that this isn't an `if else`?

Meaning is there any situation where the if statement above is true and then you expect this if statement to also be true?

If not, then this should be an `if else` otherwise the code evaluates this condition unnecessarily and is also more confusing to the reader.